Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/elm/visiting/FunctionRefEvaluator.java

```diff
@@ -0,0 +1,59 @@
+package org.opencds.cqf.cql.engine.elm.visiting;
+
+import org.hl7.elm.r1.Expression;
+import org.hl7.elm.r1.FunctionDef;
+import org.hl7.elm.r1.FunctionRef;
+import org.opencds.cqf.cql.engine.execution.CqlEngine;
+import org.opencds.cqf.cql.engine.execution.State;
+import org.opencds.cqf.cql.engine.execution.Variable;
+
+import java.util.ArrayList;
+
+public class FunctionRefEvaluator {
+
+    private FunctionDef cachedFunctionDef;
```

Comment:
Making this an instance breaks the function caching. We should make this class static, and add a  `Map<FunctionRef, FunctionDef> functionCache` the `State` object. Then this evaluator should be able to use/update the state as needed to resolve the cache.

Reply:
I mean the `Cache` object, not the `State` object.